core/arch/arm/plat-hikey/conf.mk

```diff
@@ -55,6 +55,13 @@ CFG_CORE_BGET_BESTFIT ?= y
 ifeq ($(CFG_ARM32_core),y)
 CFG_ASAN_SHADOW_OFFSET ?= 0x372E38E0
 endif
+# Hikey960 4G/6G versions have physical addresses above 4G range
+ifeq ($(CFG_DRAM_SIZE_GB),4)
+$(call force,CFG_CORE_ARM64_PA_BITS,36)
+endif
+ifeq ($(CFG_DRAM_SIZE_GB),6)
+$(call force,CFG_CORE_ARM64_PA_BITS,36)
+endif
```

Comment:
To avoid duplication:
```make
# Hikey960 4G/6G versions have physical addresses above 4G range
ifneq (,$(filter 4 6,$(CFG_DRAM_SIZE_GB)))
$(call force,CFG_CORE_ARM64_PA_BITS,36)
endif
```

Reply:
Right; did consider this but found it less readable in the end.
Will align with your recommendation...